cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/events/KubernetesRestartEventPublisher.java

```diff
@@ -37,30 +36,24 @@ public abstract class KubernetesRestartEventPublisher {
     private static final int MAX_MESSAGE_LENGTH = 1000;
     private static final String ELLIPSIS = "...";
 
-    public KubernetesRestartEventPublisher() {
-        this(Clock.systemDefaultZone());
-    }
-
-    protected KubernetesRestartEventPublisher(Clock clock) {
-        this.clock = clock;
-    }
+    private final Clock clock;
+    private final String operatorName;
+    private final KubernetesClient client;
 
     /**
-     * Create an instance for the highest event API, there's three possible subtypes, due to how fabric8 delineates the differing
-     * event APIs.
+     * Create an instance for the event API.
      *
      * @param client       Kubernetes client
      * @param operatorName the pod name of the current cluster operator instance
-     * @param hasEventsV1  if the cluster is using events.k8s.io/v1 instead of events.k8s.io/v1beta1
-     * @return instance of the appropriate publisher for the given API
      */
-    public static KubernetesRestartEventPublisher createPublisher(KubernetesClient client, String operatorName, boolean hasEventsV1) {
-        Clock clock = Clock.systemDefaultZone();
-        if (hasEventsV1) {
-            return new V1RestartEventPublisher(clock, client, operatorName);
-        } else {
-            return new V1Beta1RestartEventPublisher(clock, client, operatorName);
-        }
+    public KubernetesRestartEventPublisher(KubernetesClient client, String operatorName) {
+        this(client, operatorName, Clock.systemDefaultZone());
```

Comment:
just wondering if this is right. It's setting the clock by using the default zone of the system it's running that could be any zone around the world. At same time, we log everything using UTC. Should we use UTC clock for the events time as well?

Reply:
good catch! this was not meant to be like that.

Reply:
Ok, I understand more now, I had the impression that this is something I introduced, however after reading the code now it seems it's the way it has been so far.

I would say for this PR because it's a refactoring, let's keep it as it is, and I'd be happy to check this out in a separate PR  to be changed and tested on its own I believe.

Reply:
Fine with me, it's important to take a note of it and having an explanation. Maybe there is a good one and I am wrong. Or maybe we should really use UTC there.